include/mega/types.h

```diff
@@ -248,16 +248,15 @@ const int FOLDERNODEKEYLENGTH = 16;
 typedef list<class Sync*> sync_list;
 
 // persistent resource cache storage
-struct Cachable
+class Cacheable
 {
-    virtual bool serialize(string*) = 0;
-
-    int32_t dbid;
+public:
+    virtual ~Cacheable() = default;
```

Comment:
the destructor is an implementation detail, why not leave it at the bottom?  serialize() is the most important member

Reply:
The destructor is a very important impl. detail for base classes. You need to know if it's virtual or not to know how you can use the base class.